Makefile

```diff
@@ -25,125 +41,134 @@ else
 CLEANUP="true"
 endif
 
+# from https://suva.sh/posts/well-documented-makefiles/
+.PHONY: help
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n\nTargets:\n"} /^[0-9a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
 .PHONY: all
 all: verify test manager
 
-# Get binary dependencies
-bin/kubebuilder:
+bin/kubebuilder: ## Get binary dependencies
 	hack/download-kubebuilder-local.sh
 
-# Run all static checks
 .PHONY: verify
+verify: ## Run all static checks
 verify: verify-gomod verify-manifests verify-generate verify-protobuf verify-fmt vet
 
-# Run unit tests
 .PHONY: test
+test: ## Run unit tests
 test: bin/kubebuilder
 	KUBEBUILDER_ASSETS="$(shell pwd)/bin/kubebuilder/bin" go test ./... -coverprofile cover.out $(ARGS)
 
-# Run end to end tests in a local Kind cluster. We do not clean up after running the tests to
+.PHONY: e2e-kind
+e2e-kind: ## Run end to end tests - creates a new Kind cluster called etcd-e2e
+	go test -v -parallel ${TEST_PARALLEL_E2E} -timeout 20m ./internal/test/e2e \
+			--kind \
+			--repo-root ${CURDIR} \
+			--cleanup=${CLEANUP} \
+			--controller-image=${DOCKER_IMAGE_CONTROLLER} \
+			$(ARGS)
+# We do not clean up after running the tests to
 #  a) speed up the test run time slightly
 #  b) allow debug sessions to be attached to figure out what caused failures
-.PHONY: kind
-kind:
-	go test -parallel ${TEST_PARALLEL_E2E} -timeout 20m ./internal/test/e2e --kind --repo-root ${CURDIR} -v --cleanup=${CLEANUP} $(ARGS)
 
-# Build manager binary
+.PHONY: e2e
+e2e: ## Run the end-to-end tests - uses the current KUBE_CONFIG and context
+e2e:
+	go test -parallel ${TEST_PARALLEL_E2E} -timeout 30m ./internal/test/e2e --current-context --repo-root ${CURDIR} -v $(ARGS)
+
 .PHONY: manager
-manager:
+manager: ## Build manager binary
 	go build -o bin/manager -ldflags="-X 'github.com/improbable-eng/etcd-cluster-operator/version.Version=${VERSION}'" main.go
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
+.PHONY: run
+run: ## Run against the configured Kubernetes cluster in ~/.kube/config
+	DISABLE_WEBHOOKS=1 go run ./main.go
 # Use 'DISABLE_WEBHOOKS=1` to run the controller-manager without the
 # webhook server, and to skip the loading of webhook TLS keys, since these are
 # difficult to set up locally.
```

Comment:
Should this comment block be above the `run` block?

Reply:
Done